docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+# Cachi2 Yarn v1 (Yarn Classic) Prefetch Design
+
+## Yarn Overview
+[Yarn](https://github.com/yarnpkg/yarn) is a package manager for JavaScript. While Yarn v1.x is no longer actively developed and has been
+succeeded by [Yarn v2+ (Berry)](https://github.com/yarnpkg/berry), it [remains widely used](https://npmtrends.com/yarn).
+
+## Cachito Implementation Overview
+Cachito supports the Yarn 1.x package manager. It directly parses package.json files and uses our [pyarn](https://github.com/containerbuildsystem/pyarn)
+library to parse yarn.lock files. Dependencies are downloaded from their resolved URLs in the yarn.lock file without executing yarn itself.
+
+Registry dependencies are fetched through a proxy repository in Nexus and the user's build is configured to use that proxy as the registry.
+
+Non-registry dependencies are packed and uploaded to a hosted Nexus repository. The versions of these non-registry dependencies are
+updated in package.json and yarn.lock to match those in the hosted repository.
+
+## Proposed Cachi2 Implementation Overview
+
+### Specifying a Yarn Package in a Cachi2 Request
+Cachi2 users will specify ```yarn``` as the package manager or package type for all versions of Yarn. The presence of the ```__metadata``` field in yarn.lock
+[can be used to differentiate](https://github.com/yarnpkg/berry/blob/13d5b3041794c33171808fdce635461ff4ab5c4e/packages/yarnpkg-core/sources/Project.ts#L374)
+between Yarn v1.x and Yarn v2+ and route the request to the correct Yarn module in Cachi2.
+
+### Offline Mirror
+Prefetching dependencies for Yarn in Cachi2 will be done using Yarn's [offline mirror](https://classic.yarnpkg.com/blog/2016/11/24/offline-mirror/)
+feature. When a project is configured to use the offline mirror, Yarn will store compressed archives in the mirror directory and can install
+cached project dependencies from there later without network access.
+
+The offline mirror also has a setting that controls [whether it removes package archives](https://classic.yarnpkg.com/en/docs/prune-offline-mirror/)
+from the mirror directory when they are no longer needed. Cachi2 will want to disable automatic pruning of the offline mirror since we support multiple
+yarn projects in a single request.
+
+To enable the offline mirror, the following settings need to be applied either via .yarnrc or from environment variables:
```

Comment:
> either via .yarnrc or from environment variables:

Are we going to pick one?

Reply:
Yeah, environment variables are preferred for sure :+1:

I think I was just mentioning it for completeness here since the section about environment variables is further down, but I see how that could be confusing